### ISSUE TITLE
Intel compiler complains about FALLTHROUGH

### ISSUE THREAD

**Issue by byrnHDF** [HDFGroup/hdf5#3022]
\\src\\H5checksum.c(419,9): : warning : unannotated fall-through between switch labels [-Wimplicit-fallthrough]

because of define in H5private.h:
```
#   if defined(__clang__) || defined(__GNUC__) && __GNUC__ >= 7 && !defined(__INTEL_COMPILER)
#       define H5_ATTR_FALLTHROUGH __attribute__((fallthrough));
#   else
#       define H5_ATTR_FALLTHROUGH /* FALLTHROUGH */
#   endif
```

why is the "&& !defined(__INTEL_COMPILER)" as the compiler suggests adding the __attribute__?

**Comment by byrnHDF** [comment]
The C compiler identification is IntelLLVM 2023.1.0 with MSVC-like command-line

**Comment by hyoklee** [comment]
@derobins and clang-formatter modified the code 3 years ago.

**Comment by derobins** [comment]
Is this on Windows?

**Comment by byrnHDF** [comment]
> Is this on Windows?

yes. Seems like a minor issue - should be part of a larger issue dealing with warnings.
